bin/si-gen-cloud-control/src/spec/sockets.ts

```diff
@@ -20,7 +20,7 @@ export function createSocketFromProp(
     return socket;
   } else if (prop.metadata.writeOnly || prop.metadata.createOnly) {
     const socket = createSocket(prop.name, "input");
-    if (socket.data) {
+    if (prop.data) {
```

Comment:
Same nit as above, I don't think .data is optional

Reply:
Data is also nullable here as data is an Option in the PkgSpec